databricks_cli/dbfs/api.py

```diff
@@ -91,8 +91,18 @@ class DbfsApi(object):
     def __init__(self, api_client):
         self.client = DbfsService(api_client)
 
-    def list_files(self, dbfs_path, headers=None):
-        list_response = self.client.list(dbfs_path.absolute_path, headers=headers)
+    def _recursive_list(self, *args, **kwargs):
+        paths = self.client.list_files(*args, **kwargs)
+        files = [p for p in paths if not p.is_dir]
+        for p in paths:
+            files = files + self._recursive_list(p) if p.is_dir else files
```

Comment:
I had to read this a couple times to understand what you're doing. I think it should be simplified to iterate over `paths` just once and then have an `if p.is_dir` in there with both dealing with a file or a directory.

Reply:
Separate comment: this also need to pass along the `headers` from the `**kwargs` to be consistent with the `list_files` API today. Otherwise you use it on the first call but not later calls.